controller/managers/ControllerManager.py

```diff
@@ -182,15 +188,60 @@ def StartAutoMLProcess(self, configuration) -> Controller_pb2.StartAutoMLprocess
         Return start process status
         """
         response = Controller_pb2.StartAutoMLprocessResponse()
-        
+
+        # restructure configuration into python dictionaries
+        config = {
+            "dataset": configuration.dataset,
+            "task": configuration.task,
+            "tabular_config": {
+                "target": {
+                    "target": configuration.tabularConfig.target.target,
+                    "type": configuration.tabularConfig.target.type,
+                },
+                "features": dict(configuration.tabularConfig.features)
+            },
+            "file_configuration": dict(configuration.fileConfiguration),
+            "metric": configuration.metric,
+            "status": "running",
+            "models": [],
+            "runtime_constraints": {
+                "runtime_limit": configuration.runtimeConstraints.runtime_limit,
+                "max_iter": configuration.runtimeConstraints.max_iter
+            },
+            "required_automls": list(configuration.requiredAutoMLs),
+        }
+        username = "automl_user"
+        sess_id = self.__data_storage.insert_session(username, config)
+        print(f"inserted new session: {sess_id}")
+
         # will be called when any automl is done
-        def callback(session_id, automl_name, result):
-            print(f"automl is done: {automl_name}, session: {session_id}, result: {result}")
+        # NOTE: will run in parallel
+        def callback(session_id, model: 'dict[str, object]'):
+            _mdl_id = self.__data_storage.insert_model(username, model)
+            print(f"inserted new model: {_mdl_id}")
+
+            # lock data storage to prevent race condition between get and update
+            with self.__data_storage.lock():
+                # append new model to session
+                _sess = self.__data_storage.get_session(username, session_id)
+                self.__data_storage.update_session(username, session_id, {
+                    "models": _sess["models"] + [_mdl_id],
+                    "status": "completed"
+                })
+
+        # find requested dataset 
+        found, dataset = self.__data_storage.find_dataset(username, configuration.dataset)
+        if not found:
+            raise Exception(f"cannot find dataset with name: {configuration.dataset}")
+
+        # TODO: rework file access in AutoMLSession
+        #       we do not want to make datastore paths public
+        dataset_folder = os.path.dirname(dataset["path"])
+
+        newSession: AutoMLSession = self.__adapterManager.start_automl(configuration, dataset_folder,
```

Comment:
This will crash since `configuration` references the dataset by name, not by id.
Datasets are saved on disk by database  id. The AutoML process will look for the dataset under the wrong path.

Fix is on the way